src/main/java/jade/storage/Storage.java

```diff
@@ -0,0 +1,106 @@
+package jade.storage;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Scanner;
+
+import jade.data.Deadline;
+import jade.data.Event;
+import jade.data.Task;
+import jade.data.TaskList;
+import jade.data.Todo;
+import jade.exception.JadeException;
+
+/**
+ * The <code>Storage</code> object for loading user tasks from local file
+ * and save changes to the same file after program exits.
+ */
+public class Storage {
+    private final String filePath; // the file path storing all user tasks
+
+    /**
+     * Class constructor specifying the local file path.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Returns all saved user tasks in the local file.
+     * If the directory or file does not exit, an empty list is returned.
+     *
+     * @return A list of user tasks.
+     * @throws JadeException If directory or file does not exit
+     */
+    public List<Task> load() throws JadeException {
```

Comment:
This method felt a bit long and difficult to understand, maybe it would be better to separate it out into helper functions and the main function?